### PR TITLE
[Suggestion] - Nested setCommits & readme fixes

### PR DESCRIPTION
**Pull request by BlitZz96**
After the **beautiful** work on the pr #139 finally got released, I implemented it to my org's codebase but in the process, I felt it was missing something:

### 1. Nesting/Grouping of setCommits options
The new options: ` commit, previousCommit, repo, auto` feels a bit polluting in the sentry-webpack-plugin's "global" options scope in my opinion. The key annoying me the most is `auto`.
`auto` has no context when seen like this:
```js
const sentryCliPlugin = new SentryCliPlugin({
      release: '42',
      auto: true,
});
```
My suggestion is to group the setCommits/git options in a object:
```js
const sentryCliPlugin = new SentryCliPlugin({
      release: '42',
      setCommits: {
          auto: true,
          //Other setCommits options here...
      }
});
```
I modified the code in index.js to reflect this and tried to do the same in the tests. However, **the tests are not passing**. I have no experience with writing tests sadly, so I do not see what I can do to make them pass. Logging `this.options.setCommits` gives expecting result 🤷‍♂. Hopefully someone can help me out here if they think this PR is reasonable.

### 2. Readme changes
I do not know if there is a typo here or if I'm too tired to see that i'm wrong, but it seems to me that the key: `repo` :
https://github.com/getsentry/sentry-webpack-plugin/blob/5363d16d563709fe04291b218cfd80a629d637e1/README.md#L115

is indeed **not** optional:
https://github.com/getsentry/sentry-webpack-plugin/blob/5363d16d563709fe04291b218cfd80a629d637e1/src/index.js#L327

I don't know about the others, but I think we should do a deeper check to see what is required and what is not.

Yet again, awesome PR! I have been waiting for this for a long time ❤️

### CODE REVIEW

**Comment by BlitZz96** [comment]
Thoughts, @dmitryrn?

**Comment by kamilogorek** [comment]
I was thinking about doing that, however, I let it be as is, because this is how sentry-cli implements it – https://github.com/getsentry/sentry-cli/commit/4b913f0097acb7e896819d0f5b94f7c4f99b2bc9#diff-1f9d9d0fe3b1cdb764fff6a8b443cb7dR49-R54

It'd be breaking change now, so we'd have to allow for both versions and just document grouped one. This'd require a change here:

```js
const { commit, previousCommit, repo, auto } = (this.options.setCommits || this.options);
```

**Comment by kamilogorek** [comment]
Now that README includes a table, can you include it in this commit as well?

**Comment by BlitZz96** [comment]
> Now that README includes a table, can you include the update in this commit as well?

It would be my honor. 🛡

File: src/__tests__/index.spec.js [hunk 1]
```diff
@@ -232,10 +232,11 @@ describe('afterEmitHook', () => {
...
-      commit: '4d8656426ca13eab19581499da93408e30fdd9ef',
-      previousCommit: 'b6b0e11e74fd55836d3299cef88531b2a34c2514',
-      repo: 'group / repo',
-      // auto,
+      setCommits: {
+        commit: '4d8656426ca13eab19581499da93408e30fdd9ef',
+        previousCommit: 'b6b0e11e74fd55836d3299cef88531b2a34c2514',
+        repo: 'group / repo',
+      },
     });
```
Comment (dmitryrn) [inline src/__tests__/index.spec.js]: I would suggest to add test for both versions of calling `SentryCliPlugin({ ..., commit, previousCommit, repo })` and `SentryCliPlugin({ ..., setCommits: { commit, previousCommit, repo } })`

**Comment by kamilogorek** [comment]
[tests.patch.zip](https://github.com/getsentry/sentry-webpack-plugin/files/3707697/tests.patch.zip)

Download, unzip, `git apply <path-to-tests.patch>` :)

**Comment by kamilogorek** [comment]
Also call `yarn fix` before pushing new changes, it'll fix linting issues for you and CI will be green again

File: src/__tests__/index.spec.js (hunk 1 above)
Comment (BlitZz96) [reply src/__tests__/index.spec.js]: Thank you! I just used the "easy way" by using the patch posted below. Is there something you would like to add on that, or are we good? :D

Comment (dmitryrn) [reply src/__tests__/index.spec.js]: We are good 🚀

**Pull request merged**